notebook/data_exploration/dataset_stats.ipynb

```diff
@@ -0,0 +1,262 @@
+   "source": [
+    "# Not all labeled samples show up in the compendium, which causes pandas to panic. To fix this we have to \n",
+    "# take the intersection of the accessions in sample_ids and the accessions in the compendium\n",
+    "compendium_path = os.path.join(curr_path, os.pardir, 'data', 'human_compendium', 'HOMO_SAPIENS.tsv')\n",
+    "\n",
+    "header_ids = None\n",
+    "with open(compendium_path) as in_file:\n",
+    "    header = in_file.readline()\n",
+    "    header_ids = header.split('\\t')\n",
+    "\n",
+    "valid_sample_ids = [id_ for id_ in sample_ids if id_ in header_ids]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "metadata_file = os.path.join(curr_path, os.pardir, 'data', 'all_metadata.json')\n",
+    "metadata_json = json.load(open(metadata_file))\n",
+    "sample_metadata = metadata_json['samples']\n",
+    "\n",
+    "sample_to_study = {}\n",
+    "\n",
+    "experiments = metadata_json['experiments']\n",
+    "\n",
+    "for study in experiments:\n",
+    "    for accession in experiments[study]['sample_accession_codes']:\n",
+    "        sample_to_study[accession] = study"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Remove beadchip info as it will be removed on its own later\n",
+    "no_beadchip_ids = []\n",
+    "for sample in valid_sample_ids:\n",
+    "    if 'beadchip' not in sample_metadata[sample]['refinebio_platform'].lower():\n",
+    "        # Leave out 'other' labels as they won't be included in the analysis\n",
+    "        if sample_to_label[sample] != 'other':\n",
+    "            no_beadchip_ids.append(sample)\n",
+    "valid_sample_ids = no_beadchip_ids"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "studies = set()\n",
+    "platforms = []\n",
+    "diseases = []\n",
+    "disease_study_counts = {}\n",
+    "\n",
+    "for sample in valid_sample_ids:\n",
+    "    study = sample_to_study[sample]\n",
+    "    studies.add(study)\n",
+    "    platform = sample_metadata[sample]['refinebio_platform'].lower()\n",
+    "    platforms.append(platform)\n",
+    "    \n",
+    "    disease = sample_to_label[sample]\n",
+    "    diseases.append(disease)\n",
+    "    \n",
+    "    if disease in disease_study_counts:\n",
+    "        if study not in disease_study_counts[disease]:\n",
+    "            disease_study_counts[disease].append(study)\n",
+    "    else:\n",
+    "        disease_study_counts[disease] = [study]\n",
+    "        \n",
+    "\n",
+    "platforms = Counter(platforms)\n",
+    "diseases = Counter(diseases)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id='stats'></a>\n",
+    "## Stats\n",
+    "All code beyond this point exists solely to print statistics about the data"
```

Comment:
I haven't heard of "Counter" before, this is the nice function that i'll have to keep in mind!

Just for my own curiosity, I'm not sure if you're planning to combine everything but it looks like you'll have overrepresentations of certain platforms and diseases, I wonder how this will effect your analysis.  If you're planning to choose a specific platform and disease, what are the counts for that combination?

Reply:
Yeah, it's very useful! There's a lot of cool stuff in python's `collections` module

I will probably leave out less studies diseases and less used platforms, but haven't decided what the cutoffs will be. I guess I don't have an answer for this quite yet, sorry